File: src/main/java/com/devspace/conexfy/services/ConConnectionService.java
package com.devspace.conexfy.services;

import com.devspace.conexfy.entities.ConConnectionEntity;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ConConnectionService {
    Flux<ConConnectionEntity> findAll();
    Mono<ConConnectionEntity> findById(Long id);
    Mono<ConConnectionEntity> create(ConConnectionEntity entity);
    Mono<ConConnectionEntity> update(ConConnectionEntity entity);
    Mono<Void> deleteById(Long id);
    Mono<String> execute(Long id);
}

File: src/main/kotlin/com/devspace/conexfy/services/ConConnectionService.kt
package com.devspace.conexfy.services

import com.devspace.conexfy.entities.ConConnectionEntity
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface ConConnectionService {
    fun findAll(): Flux<ConConnectionEntity>
    fun findById(id: Long): Mono<ConConnectionEntity>
    fun create(entity: ConConnectionEntity): Mono<ConConnectionEntity>
    fun update(entity: ConConnectionEntity): Mono<ConConnectionEntity>
    fun deleteById(id: Long): Mono<Void>
    fun execute(id: Long): Mono<String>
}
